salt/cloud/clouds/vultrpy.py

```diff
@@ -624,28 +745,45 @@ def _query(path, method="GET", data=None, params=None, header_dict=None, decode=
         search_global=False,
         default="api.vultr.com",
     )
-    url = "https://{management_host}/v1/{path}?api_key={api_key}".format(
+    url = "https://{management_host}/v2/{path}".format(
         management_host=management_host,
         path=path,
-        api_key=api_key,
     )
 
-    if header_dict is None:
-        header_dict = {}
-
-    result = __utils__["http.query"](
+    data = salt.utils.json.dumps(data)
+    requester = getattr(requests, method.lower())
+    # pylint: disable=not-callable
+    request = requester(
         url,
-        method=method,
-        params=params,
         data=data,
-        header_dict=header_dict,
-        port=443,
-        text=True,
-        decode=decode,
-        decode_type="json",
-        hide_fields=["api_key"],
-        opts=__opts__,
+        headers={
+            "Authorization": "Bearer " + api_key,
+            "Content-Type": "application/json",
+        },
     )
+    # pylint: enable=not-callable
+    if request.status_code > 299:
+        raise SaltCloudSystemExit(
+            "An error occurred while querying Vultr. HTTP Code: {}  "
+            "Error: '{}'".format(
+                request.status_code,
+                # request.read()
+                request.text,
+            )
+        )
+
+    log.debug(request.url)
```

Comment:
Would it be possible for this url to contain sensitive information, such as a password? If so, I don't think we should log it

Also, I think we need more information around this log. Possibly `log.debug("Querying url: {}".format(request.url))`

Reply:
iirc there should be no sensitive information, I'd have to take another look at the code as it's now quite a while ago that I wrote it ;)

iirc tokens & such are only transmitted via headers, so they wouldn't be logged here. 

It's possible SSH public keys could be set as URL params (iirc they're json data posted to an API endpoint, so again should be safe).
Besides that I don't think there could be sensitive information, unless we consider machine ID's and such as sensitive.

I'll take an critical look at it when I get back to it & modify the debug statement as requested.